client/resource_group/controller/controller.go

```diff
@@ -491,31 +506,14 @@ func (gc *groupCostController) initRunState() {
 	}
 }
 
-func (gc *groupCostController) updateRunState(ctx context.Context) {
+func (gc *groupCostController) updateRunState() {
 	newTime := time.Now()
-	deltaConsumption := &rmpb.Consumption{}
+	gc.mu.Lock()
 	for _, calc := range gc.calculators {
-		calc.Trickle(ctx, deltaConsumption)
+		calc.Trickle(gc.mu.consumption, gc.mainCfg.IsSingleGroupByKeyspace)
```

Comment:
Considering it will hold the lock now, will `Trickle` cost a long time to process here to make the locking time unexpectedly long?

Reply:
done.

Reply:
- We need to get the previous sql cpu time to calculate the delta consumption, so we need to hold lock to pass the race test.
-  For now impl, Trickle doesn't take long time to get the SQL time.